zookeeper-api/src/main/java/org/apache/helix/zookeeper/zkclient/callback/ZkAsyncCallbacks.java

```diff
@@ -50,6 +50,10 @@ public void processResult(int rc, String path, Object ctx, byte[] data, Stat sta
         if (_data != null && ctx != null && ctx instanceof ZkAsyncCallMonitorContext) {
           ((ZkAsyncCallMonitorContext) ctx).setBytes(_data.length);
         }
+      } else if(rc != Code.NONODE.intValue()) {
+        if (ctx instanceof ZkAsyncCallMonitorContext) {
```

Comment:
We could condense the two if lines into one && condition.

Reply:
Thanks, I'll do it in the next commit.